File: cpp/leetcode/160.cpp
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        if(!headA || !headB) return NULL;
        ListNode *pa = headA, *pb = headB;
        while(pa != pb){
            pa = !pa? headB: pa->next;
            pb = !pb? headA: pb->next;
        }
        return pa;
    }
};

File: python/leetcode/160.py
class Solution:
    def getIntersectionNode(self, headA, headB):
        if not headA or not headB: return None
        pa, pb = headA, headB
        while pa is not pb:
            pa = headB if not pa else pa.next
            pb = headA if not pb else pb.next
        return pa  